.github/workflows/ruby-ci.yml

```diff
@@ -0,0 +1,17 @@
+name: Ruby-CI
+on: [pull_request]
+permissions:
+  contents: read
+
+jobs:
+  test:
```

Comment:
You will need to add the service of PostgreSQL here as well, e.g: https://github.com/theforeman/foreman_ansible/blob/master/.github/workflows/ruby_tests.yml#L24

and after that you need to run the DB preparation commands:
https://github.com/theforeman/foreman_ansible/blob/master/.github/workflows/ruby_tests.yml#L66-L68

if this is too much it can be on another PR.

Reply:
Created this new issue https://github.com/girijaasoni/newsletterApplicaton/issues/23 for the same. Will update the CI test workflow there.